Next-formalizationFull.hs

```diff
@@ -186,10 +229,65 @@ afterClosed (d, m, y) func int evt = not (beforeOpen (d, m, y) func int evt)
 inperiod:: Date -> (EventInstance -> TimeStamps) -> Int -> Date -> (EventInstance -> TimeStamps) -> Int -> EventInstance -> Bool
 inperiod t1 func1 int1 t2 func2 int2 evt = afterClosed t1 func1 int1 evt && beforeClosed t2 func2 int2 evt
 
+-- is done
+isdone:: EventInstance -> Bool
+isdone e = if List.genericLength (getActualDate e) == 3 then True else False
```

Comment:
Do you mean should I add an explanation for isdone function or something else? isdone function looks at the actualDate field of an event instance and determines whether the event instance has a Done date.